source/D2CommonDefinitions/include/D2PacketDef.h

```diff
@@ -298,13 +298,18 @@ struct D2GSPacketClt31		//size of 0x09
 
 struct D2GSPacketClt32		//size of 0x11
 {
-	uint8_t nHeader;			//0x00
-	int32_t unk0x01;			//0x01
-	int32_t unk0x05;			//0x05
-	uint16_t unk0x09;			//0x09
-	uint16_t unk0x0B;			//0x0B
-	int32_t unk0x0D;			//0x0D
+	uint8_t nHeader;						//0x00
+	int32_t dwNpcGUID;						//0x01
+	int32_t dwItemGUID;						//0x05
+	uint16_t nTransactionType;              //0x09
+	struct {                                //0x0B
+		uint8_t nItemMode : 8;
+		uint8_t : 7;
+		uint8_t bFill : 1;
```

Comment:
It's a bit annoying that I can't do the following:

```
	struct {                                //0x0B
		uint16_t nItemMode : 15;
		uint8_t bFill : 1;
	};
```
seems the compiler insists on padding the uint16_t to 16 bits, at which point bFill causes an extra byte.

I don't think the 7 bits are actually used for anything, so I'm just dropping them on the floor.

Reply:
This is because you are changing the type.
I'm guessing 

```c++
struct {                                //0x0B
		uint16_t nItemMode : 15;
		uint16_t bFill : 1;
	};
```
would work.

Reply:
Learned something new. I could also now do something like:
```
	struct {
		D2C_TransactionTypes nTransactionType : 15;
		D2C_ItemModes nItemMode : 15;
		uint32_t bFill : 1;
	};
```
and then change signatures of D2GAME_NPC_BuyItemHandler_6FCC92A0 and below to use enum types that are bigger.
In reality the original functions have 2 byte types, but I think passing 4 bytes values might still work?
Perhaps it might not work if the values are passed on the stack depending on the calling convention?

Reply:
or maybe it would be ok:
https://stackoverflow.com/questions/45127993/how-many-bytes-does-the-push-instruction-push-onto-the-stack-when-i-dont-specif/45128435#45128435

Reply:
I think the current version is good enough, putting enums in a bitfield will be ambigous/risky (different types might trigger new bitfield). I think it is good to have the `uint16_t` so that it is explicit that we are packing something unrelated into 16bits.